environment/src/main/java/com/sequenceiq/environment/environment/dto/EnvironmentDtoConverter.java

```diff
@@ -204,21 +200,17 @@ public Environment creationDtoToEnvironment(EnvironmentCreationDto creationDto)
         environment.setCreated(System.currentTimeMillis());
         environment.setTags(getTags(creationDto));
         environment.setExperimentalFeaturesJson(creationDto.getExperimentalFeatures());
-        setLocation(creationDto, environment, location);
+        setRegions(creationDto, environment);
         return environment;
     }
 
-    private void setLocation(EnvironmentCreationDto creationDto, Environment environment, LocationDto location) {
```

Comment:
Isn't the `Location` used by the UI?

Reply:
It is, even the UT checks for this. :smile:  This method is called from `creationDtoToEnvironment()`, and the latter already populates the location-related fields in lines 183-189:
```
        LocationDto location = creationDto.getLocation();
        environment.setLatitude(location.getLatitude());
        environment.setLongitude(location.getLongitude());
        environment.setLocation(location.getName());
        environment.setTelemetry(creationDto.getTelemetry());
        environment.setBackup(creationDto.getBackup());
        environment.setLocationDisplayName(location.getDisplayName());
```
Hence I opted for refactoring `setLocation()` so that it only needs to deal with regions.